### PR TITLE
feat: solve outstanding TODOs

### PR DESCRIPTION
**Pull request by merklefruit**
starts #359 

* feat(sidecar): added a robust generic future retry system
* feat(sidecar): removed usage of validator indexes, in favor of validator pubkeys
* chore(sidecar): remove TODOs in primitives module (nits, docs, and reminders)

Note: accompanying kurtosis PR: https://github.com/chainbound/ethereum-package/pull/24

### CODE REVIEW

File: bolt-sidecar/src/common.rs
Comment (thedevbirb) [inline bolt-sidecar/src/common.rs:1]: This is cool, and nice tests!

**Review by thedevbirb — APPROVED** [review]
Great refactor! Thank you

**Comment by thedevbirb** [comment]
Can you fix CI?

**Comment by merklefruit** [comment]
> Can you fix CI?

Since it's just linkspector failing on links that are unrelated to this PR (in bolt-contracts), I suggest fixing them on holesky-launch directly

**Pull request merged**